### ISSUE TITLE
Migrate to Java 8

### ISSUE THREAD

**Issue by clementdenis** [cloudendpoints/endpoints-java#162]
With [the shutdown of Java 7 App Engine standard runtime](https://cloud.google.com/appengine/docs/deprecations/java7) coming soon, what about upgrading the code base of Cloud Endpoints framework to Java 8?

I'm available to help :-)

**Comment by tangiel** [comment]
Yeah, I will probably work on that. Will also want to bump the Guava version, which actually involves changing the artifact to a new one, as Guava now has different artifacts for Java 7- and Java 8+. At that point I will probably bump the minor version.
